plugins/registry.js

```diff
@@ -49,6 +49,39 @@ const resolveReactElement = (element) => {
     return element;
 };
 
+class SubMenuGroup {
```

Comment:
I think the class is only being used to try to manage the internal state, but a higher-order function seems like a more natural fit here. If we converge on some of the proposals below to use the same action type and treat an empty `subMenu` as just a normal case of a menu item without a submenu, I think we can eliminate the duplicate code as well.

Reply:
Converted the class to a higher-order function, inside of `registerPostDropdownSubMenuItem`.  It looks much cleaner, thank you.

Reply:
I believe all changes requested are addressed.